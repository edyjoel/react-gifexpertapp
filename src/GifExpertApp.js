import React, { useState } from 'react';
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({defaultCategories = [] }) => {

  const [categories, setCategories] = useState(defaultCategories);

  // const handleAdd = () => {
  //   // setCategories( [...categories, 'HunterXHunder'] );
  //   // setCategories( ['HunterXHunder', ...categories] );
  //   setCategories( cats => [...cats, 'HunterXHunder'] );
  // }

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory setCategories={ setCategories } />
      <hr/>


      <ol>
        {
          categories.map((category, i) => (
            <GifGrid key={ category } category={category } />
          ))
        }
      </ol>
    </>
  )
}

export default GifExpertApp;
