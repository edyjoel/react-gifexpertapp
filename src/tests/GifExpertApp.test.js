import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas con <GifExpertApp />', () => {
  test('debe de mostrar el componente correctamente', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });
  test('debe de mostar una lista de categorias', () => {
    const categories = ['one punch', 'Dragon ball'];
    const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});