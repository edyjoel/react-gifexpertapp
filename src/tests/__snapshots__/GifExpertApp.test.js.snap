// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con <GifExpertApp /> debe de mostar una lista de categorias 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="one punch"
      key="one punch"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas con <GifExpertApp /> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
